app/src/main/java/com/github/geohunt/app/MainActivity.kt

```diff
@@ -36,7 +38,11 @@ class MainActivity : ComponentActivity() {
 
         setContent {
             GeoHuntScreen {
-                MainScreen(database, viewModel)
+                MainScreen(database, viewModel) {
+                    viewModel.logout(this@MainActivity) {
+                        replaceActivity(Intent(this@MainActivity, LoginActivity::class.java))
+                    }
+                }
```

Comment:
I would advocate for using named parameters instead of a trailing lambda, as it might be confusing as to what that represents. Something like:
```suggestion
                MainScreen(database, viewModel, logout = {
                    viewModel.logout(this@MainActivity, then = {
                        replaceActivity(Intent(this@MainActivity, LoginActivity::class.java))
                    })
                })
```